test/kinesis/testHooks-error.test.js

```diff
@@ -0,0 +1,21 @@
+const test = require('ava');
+const sinon = require('sinon');
+
+test('The afterAll hook handles errors in the beforeAll hook gracefully', async (test) => {
+  // Stub the docker module to throw errors when fetching images.
+  // This needs to happen before the dynamodb helper module is imported
+  const docker = require('../../src/docker');
+  const error = new Error('Stubbed failure');
+  const ensureStub = sinon.stub(docker, 'ensureImage')
+    .rejects(error);
+
+  const { kinesisTestHooks } = require('../../src/kinesis');
+  const { afterAll, beforeAll } = kinesisTestHooks(false);
+
+  try {
+    await test.throwsAsync(beforeAll, { instanceOf: Error, message: error.message });
+    await afterAll();
```

Comment:
Minor: I'm assuming the point if this test is that the `afterAll()` function does not throw an error. Ava has a `notThrowsAsync` that might make this more explicit.

Reply:
Nice, I'll add that.